### ISSUE TITLE
Intermittent connectivity on Egress only NetworkPolicy

### ISSUE THREAD

**Issue by SushiFu** [aws/aws-network-policy-agent#77]
Hi 👋 ,

**Context**

I'm trying to use a `NetworkPolicy` (below) which enforces Egress rules only.
I want my Jobs to be able to reach Internet freely and be only able to reach `coredns` and an `api` on the private space.
My Jobs are talking to the `api` though GRPC.

**Problem**

I'm scheduling a fair amount of containers (~100) some containers are able to reach `coredns` and the `api` and some others don't.
Even weirder some Jobs begin communication with the `api` but failed to reach again after some times.

```
---
kind: NetworkPolicy
apiVersion: networking.k8s.io/v1
metadata:
  name: tasks
  namespace: tasks
spec:
  podSelector: {}
  ingress:
    - {}
  egress:
    # IPV4 except RFC1918
    - to:
        - ipBlock:
            cidr: 0.0.0.0/1
            except:
              - 10.0.0.0/8
        - ipBlock:
            cidr: 128.0.0.0/1
            except:
              - 172.16.0.0/12
              - 192.168.0.0/16
    - to:
        # allow kube-dns
        - namespaceSelector:
            matchLabels:
              kubernetes.io/metadata.name: kube-system
          podSelector:
            matchLabels:
              k8s-app: kube-dns
      ports:
        - port: 53
          protocol: UDP
    - to:
        # allow api
        - namespaceSelector:
            matchLabels:
              kubernetes.io/metadata.name: api
          podSelector:
            matchLabels:
              app: api
      ports:
        - protocol: TCP
          port: 8000
```

Here are some logs from `/var/log/aws-routed-eni/network-policy-agent.log` which demonstrate:
```
{"level":"info","timestamp":"2023-09-27T11:37:56.947Z","logger":"controllers.policyEndpoints","msg":"Total number of PolicyEndpoint resources for","podIdentifier ":"task-6c172eb8-c201-4374-b070-c19215333aac-tasks"," are ":2}
{"level":"info","timestamp":"2023-09-27T11:37:56.947Z","logger":"controllers.policyEndpoints","msg":"Default Deny enabled on Ingress"}
{"level":"info","timestamp":"2023-09-27T11:37:56.947Z","logger":"ebpf-client","msg":"Pod already has Ingress Probe attached - ","Name: ":"task-6c172eb8-c201-4374-b070-c19215333aac-hvds4","Namespace: ":"tasks"}
{"level":"info","timestamp":"2023-09-27T11:37:56.947Z","logger":"ebpf-client","msg":"Pod already has Egress Probe attached - ","Name: ":"task-6c172eb8-c201-4374-b070-c19215333aac-hvds4","Namespace: ":"tasks"}
{"level":"info","timestamp":"2023-09-27T11:37:56.947Z","logger":"ebpf-client","msg":"AttacheBPFProbes for","pod":"task-6c172eb8-c201-4374-b070-c19215333aac-hvds4"," in namespace":"tasks"," with hostVethName":"eni6dcc57d426e"}
{"level":"info","timestamp":"2023-09-27T11:37:56.947Z","logger":"controllers.policyEndpoints","msg":"Successfully attached required eBPF probes for","pod:":"task-6c172eb8-c201-4374-b070-c19215333aac-hvds4","in namespace":"tasks"}
{"level":"info","timestamp":"2023-09-27T11:37:56.947Z","logger":"controllers.policyEndpoints","msg":"Target Pod doesn't belong to the current pod Identifier: ","Name: ":"task-bc07e709-a28d-46e8-ac85-58ebb60b9f87-k922d","Pod ID: ":"task-6c172eb8-c201-4374-b070-c19215333aac-tasks"}
{"level":"info","timestamp":"2023-09-27T11:37:56.947Z","logger":"ebpf-client","msg":"Pod has an Ingress hook attached. Update the corresponding map","progFD: ":97,"mapName: ":"ingress_map"}
{"level":"info","timestamp":"2023-09-27T11:37:56.947Z","logger":"ebpf-client","msg":"Pod has an Egress hook attached. Update the corresponding map","progFD: ":99,"mapName: ":"egress_map"}
{"level":"info","timestamp":"2023-09-27T11:37:58.011Z","logger":"ebpf-client","msg":"Flow Info:  ","Src IP":"10.72.58.134","Src Port":56358,"Dest IP":"172.20.0.10","Dest Port":53,"Proto":"UDP","Verdict":"ACCEPT"}
{"level":"info","timestamp":"2023-09-27T11:37:58.011Z","logger":"ebpf-client","msg":"Flow Info:  ","Src IP":"10.72.58.134","Src Port":52378,"Dest IP":"172.20.0.10","Dest Port":53,"Proto":"UDP","Verdict":"ACCEPT"}
{"level":"info","timestamp":"2023-09-27T11:37:58.012Z","logger":"ebpf-client","msg":"Flow Info:  ","Src IP":"10.72.58.134","Src Port":49799,"Dest IP":"172.20.0.10","Dest Port":53,"Proto":"UDP","Verdict":"ACCEPT"}
[...]
{"level":"info","timestamp":"2023-09-27T10:42:15.679Z","logger":"controllers.policyEndpoints","msg":"Total number of PolicyEndpoint resources for","podIdentifier ":"task-289bda54-39e5-4818-9018-0f26cee1b015-tasks"," are ":2}
{"level":"info","timestamp":"2023-09-27T10:42:15.679Z","logger":"controllers.policyEndpoints","msg":"Deriving Firewall rules for PolicyEndpoint:","Name: ":"egress-allow-internet-only-4446c"}
{"level":"info","timestamp":"2023-09-27T10:42:15.679Z","logger":"controllers.policyEndpoints","msg":"Total no.of - ","ingressRules":0,"egressRules":11}
{"level":"info","timestamp":"2023-09-27T10:42:15.679Z","logger":"controllers.policyEndpoints","msg":"Deriving Firewall rules for PolicyEndpoint:","Name: ":"deny-all-qdmk2"}
{"level":"info","timestamp":"2023-09-27T10:42:15.679Z","logger":"controllers.policyEndpoints","msg":"Total no.of - ","ingressRules":0,"egressRules":11}
{"level":"info","timestamp":"2023-09-27T10:42:15.679Z","logger":"controllers.policyEndpoints","msg":"Default Deny enabled on Ingress"}
{"level":"info","timestamp":"2023-09-27T10:42:15.679Z","logger":"ebpf-client","msg":"Pod already has Ingress Probe attached - ","Name: ":"task-289bda54-39e5-4818-9018-0f26cee1b015-czq54","Namespace: ":"tasks"}
{"level":"info","timestamp":"2023-09-27T10:42:15.679Z","logger":"ebpf-client","msg":"Pod already has Egress Probe attached - ","Name: ":"task-289bda54-39e5-4818-9018-0f26cee1b015-czq54","Namespace: ":"tasks"}
{"level":"info","timestamp":"2023-09-27T10:42:15.679Z","logger":"ebpf-client","msg":"AttacheBPFProbes for","pod":"task-289bda54-39e5-4818-9018-0f26cee1b015-czq54"," in namespace":"tasks"," with hostVethName":"enicc017909cf0"}
{"level":"info","timestamp":"2023-09-27T10:42:15.679Z","logger":"controllers.policyEndpoints","msg":"Successfully attached required eBPF probes for","pod:":"task-289bda54-39e5-4818-9018-0f26cee1b015-czq54","in namespace":"tasks"}
{"level":"info","timestamp":"2023-09-27T10:42:15.679Z","logger":"ebpf-client","msg":"Pod has an Ingress hook attached. Update the corresponding map","progFD: ":49,"mapName: ":"ingress_map"}
{"level":"info","timestamp":"2023-09-27T10:42:15.679Z","logger":"ebpf-client","msg":"Total L4 entry count for catch all entry: ","count: ":0}
{"level":"info","timestamp":"2023-09-27T10:42:15.679Z","logger":"ebpf-client","msg":"ID of map to update: ","ID: ":23}
{"level":"info","timestamp":"2023-09-27T10:42:15.679Z","logger":"ebpf-client","msg":"Pod has an Egress hook attached. Update the corresponding map","progFD: ":51,"mapName: ":"egress_map"}
{"level":"info","timestamp":"2023-09-27T10:42:15.679Z","logger":"ebpf-client","msg":"Total L4 entry count for catch all entry: ","count: ":0}
{"level":"info","timestamp":"2023-09-27T10:42:15.679Z","logger":"ebpf-client","msg":"Parsed Except CIDR","IP Key: ":"10.0.0.0/8"}
{"level":"info","timestamp":"2023-09-27T10:42:15.679Z","logger":"ebpf-client","msg":"Parsed Except CIDR","IP Key: ":"172.16.0.0/12"}
{"level":"info","timestamp":"2023-09-27T10:42:15.679Z","logger":"ebpf-client","msg":"Parsed Except CIDR","IP Key: ":"192.168.0.0/16"}
{"level":"info","timestamp":"2023-09-27T10:42:15.679Z","logger":"ebpf-client","msg":"ID of map to update: ","ID: ":24}
{"level":"info","timestamp":"2023-09-27T10:42:15.779Z","logger":"ebpf-client","msg":"Flow Info:  ","Src IP":"10.72.82.5","Src Port":56885,"Dest IP":"169.254.169.253","Dest Port":53,"Proto":"UDP","Verdict":"ACCEPT"}
{"level":"info","timestamp":"2023-09-27T10:42:15.892Z","logger":"ebpf-client","msg":"Flow Info:  ","Src IP":"10.72.82.5","Src Port":36647,"Dest IP":"169.254.169.253","Dest Port":53,"Proto":"UDP","Verdict":"ACCEPT"}
{"level":"info","timestamp":"2023-09-27T10:42:15.998Z","logger":"ebpf-client","msg":"Flow Info:  ","Src IP":"10.72.75.220","Src Port":46367,"Dest IP":"172.20.0.10","Dest Port":53,"Proto":"UDP","Verdict":"DENY"}
```

**Environment:**
EKS: v1.27.5-eks-43840fb
VPC CNI: v1.15.0-eksbuild.2
VPC CNI Config: `{"enableNetworkPolicy":"true"}`
AMI: amazon/amazon-eks-node-1.27-v20230919

**Comment by jayanthvn** [comment]
@SushiFu - Sorry for the delay.

Based on the context - 

> I’m scheduling a fair amount of containers (~100) some containers are able to reach coredns and the api and some others don’t.

Probably you are running into this issue - https://github.com/aws/aws-network-policy-agent/pull/106. This is fixed with v1.0.5 network policy agent which is part of v1.15.3 release - https://github.com/aws/amazon-vpc-cni-k8s/releases/tag/v1.15.3. Can you please try with the latest?

**Comment by SushiFu** [comment]
Hi @jayanthvn,
I just made the same test in our scenario using the VPC CNI v1.15.3 release.
Indeed #106 seems to have fixed our connectivity issue.
Thanks for you help!
